fixlib/fixlib/baseresources.py

```diff
@@ -114,8 +114,71 @@ def get(self) -> Dict[str, Any]:
         return changes
 
 
-MetricName = str
-StatName = str
+# todo: replace to StrEnum once resoto is on 3.11
+class MetricName(str, Enum):
+    def __str__(self) -> str:
+        return self.value
+
+    # ec2
```

Comment:
the comment names are all AWS specific. Consider picking general-purpose names.  
ec2 is a complex service - you are probably referring to ec2 instance.